feat: React Native 0.71.14 with Xcode 15 stable

leotm opened the pull request:
**Development & PR Process**
1. Follow MetaMask [Mobile Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/coding_guidelines/CODING_GUIDELINES.md)
2. Add `release-xx` label to identify the PR slated for a upcoming release (will be used in release discussion)
3. Add `needs-dev-review` label when work is completed
4. Add the appropiate QA label when dev review is completed
    - `needs-qa`: PR requires manual QA.
    - `No QA/E2E only`: PR does not require any manual QA effort. Prior to merging, ensure that you have successful end-to-end test runs in Bitrise. 
    - `Spot check on release build`: PR does not require feature QA but needs non-automated verification. In the description section, provide test scenarios. Add screenshots, and or recordings of what was tested.
5. Add `QA Passed` label when QA has signed off (Only required if the PR was labeled with `needs-qa`)
6. Add your team's label, i.e. label starting with `team-` (or `external-contributor` label if your not a MetaMask employee)

**Description**

Investigation into Xcode 14.2+ support
- https://github.com/MetaMask/metamask-mobile/issues/6158#issuecomment-1701340482

TODO
- [x] Fix Branch build error
  - Fork Branch repo, bump `IPHONEOS_DEPLOYMENT_TARGET`s from 8 to 12, update git submodule
- [x] Fix MixPanel build error
  - Fork MixPanel repo, `IPHONEOS_DEPLOYMENT_TARGET`s from 8 to 12, update git submodule
- [x] Upgrade RN from 0.71.6 to 0.71.13
- [x] Refactor Podfile
- [x] Raise/fix Sentry build issue
- [x] Update `react-native.config.js` to support `NO_FLIPPER` env var
- [x] Fix Yoga/YogaKit build error (disable Flipper natively)
- [x] Fix Xcode 15 build error: Unable to load ...frameworks-Debug-input-files.xcfilelist
- [x] Bump BitRise stack from osx-xcode-14.2.x-ventura to osx-xcode-15.0.x
- [x] Upgrade RN from 0.71.13 to 0.71.14 for additional Xcode 15 compat
- [x] Upgrade Detox for additional Xcode 15 compat
- [x] Fix BitRise osx-xcode-15.0.x smoke tests
  - Bump Detox config iOS sim from iPhone 12 Pro to iPhone 13 Pro
- [x] Test older Xcode 14.2
- [x] Remove Flipper from release-mode
- [x] Fix Flipper debug-mode tech debt
- [x] Reenable Flipper
- [x] Update build scripts for iPhone 13 Pro
- [x] Update docs for Xcode 15 and iPhone 13 Pro

_Xcode 15 beta 8 tested up to a7ec533, then Xcode 15 stable since_

`start:ios` must be run before running via Xcode 15, or `cp node_modules/@metamask/mobile-provider/dist/index.js app/core/InpageBridgeWeb3.js` can optionally be run

_Xcode updates: https://developer.apple.com/documentation/updates/xcode_

**Screenshots/Recordings**

in link above

**Issue**

fixes https://github.com/MetaMask/metamask-mobile/issues/6158

also fixes ⚠️

<img width="576" alt="Xcode 14 2 warning" src="https://github.com/MetaMask/metamask-mobile/assets/1881059/059988b2-e90c-4400-8f6a-e015391d4d48">

**Checklist**

* [x] There is a related GitHub issue
* [ ] Tests are included if applicable
* [x] Any added code is fully documented

leotm committed:
Bump Branch IPHONEOS_DEPLOYMENT_TARGETs from 8 to 12

Fix iOS build (clang) error:
https://github.com/CocoaPods/CocoaPods/issues/12033

leotm committed:
Bump MixPanel IPHONEOS_DEPLOYMENT_TARGETs from 8 to 12

Fix iOS build (clang) error:
https://github.com/CocoaPods/CocoaPods/issues/12033

leotm committed:
Upgrade RN from 0.71.6 to 0.71.13 (npm)

leotm committed:
Disable Flipper in Podfile and update lockfile (RN 0.71.13)

- use_react_native
- post_install

leotm committed:
Add NO_FLIPPER=1 to pod:install script

leotm committed:
Account for NO_FLIPPER (react-native-flipper) in react-native.config.js

leotm committed:
Update Sentry (npm)

- update deps
- remove old resolutions

No changes to Pod lockfile, so iOS build error (SentryPrivate) sadly persists

leotm commented:
TODO: check why/how https://github.com/MetaMask/metamask-mobile/pull/7114/commits/a7ec533c98c03bff3b7f700b19bf646d671e5178 didn't trigger Pod lockfile changes spotted in
- https://github.com/MetaMask/metamask-mobile/pull/7277

leotm committed:
Revert "Update Sentry (npm)"

This reverts commit a7ec533c98c03bff3b7f700b19bf646d671e5178.

leotm committed:
Merge branch 'main' into react-native-0.71.13-xcode-15-beta-8

leotm committed:
Bump RN patch from 0.71.6 to 0.71.13

leotm committed:
Revert "Bump MixPanel IPHONEOS_DEPLOYMENT_TARGETs from 8 to 12"

This reverts commit acb6d40c94db4e3fe5d56f0ee8747f842bed8f9f.

leotm committed:
Revert "Bump Branch IPHONEOS_DEPLOYMENT_TARGETs from 8 to 12"

This reverts commit 0d199d7e8efb1c2e8f6988fe9092c8a5fae795bf.

leotm committed:
pod repo update && pod install

leotm committed:
Fix Xcode 15 error: Unable to load ...frameworks-Debug-input-files.xcfilelist

error: Unable to load contents of file list: '.../ios/Pods/Target Support Files/Pods-MetaMask/Pods-MetaMask-frameworks-Debug-input-files.xcfilelist' (in target 'MetaMask' from project 'MetaMask')

leotm commented on ios/MetaMask.xcodeproj/project.pbxproj:
```diff
@@ -945,23 +926,6 @@
 			shellPath = /bin/sh;
 			shellScript = "if [[ -s \"$HOME/.nvm/nvm.sh\" ]]; then\n. \"$HOME/.nvm/nvm.sh\"\nelif [[ -x \"$(command -v brew)\" && -s \"$(brew --prefix nvm)/nvm.sh\" ]]; then\n. \"$(brew --prefix nvm)/nvm.sh\"\nfi\n\nexport NODE_BINARY=$(which node)\n\nif [[ -s \"$HOME/.nvm/nvm.sh\" ]]; then\n. \"$HOME/.nvm/nvm.sh\"\nelif [[ -x \"$(command -v brew)\" && -s \"$(brew --prefix nvm)/nvm.sh\" ]]; then\n. \"$(br ...
 		};
-		C9BADD20B98A4D13195A3200 /* [CP] Embed Pods Frameworks */ = {
-			isa = PBXShellScriptBuildPhase;
-			buildActionMask = 2147483647;
-			files = (
-			);
-			inputFileListPaths = (
-				"${PODS_ROOT}/Target Support Files/Pods-MetaMask-QA/Pods-MetaMask-QA-frameworks-${CONFIGURATION}-input-files.xcfilelist",
-			);
-			name = "[CP] Embed Pods Frameworks";
-			outputFileListPaths = (
-				"${PODS_ROOT}/Target Support Files/Pods-MetaMask-QA/Pods-MetaMask-QA-frameworks-${CONFIGURATION}-output-files.xcfilelist",
-			);
-			runOnlyForDeploymentPostprocessing = 0;
-			shellPath = /bin/sh;
-			shellScript = "\"${PODS_ROOT}/Target Support Files/Pods-MetaMask-QA/Pods-MetaMask-QA-frameworks.sh\"\n";
-			showEnvVarsInLog = 0;
-		};
```
fixed in: https://github.com/MetaMask/metamask-mobile/pull/7114/commits/a09d517135af02474bbef3e461a7e50d32f99830

leotm replied:
fix reverted later in https://github.com/MetaMask/metamask-mobile/pull/7114/commits/8bb0e49c4f6fe01c3b6f679581488f5612fc26b5

in favour of cleaner solution in latest commits (reenabling Flipper config but disabling via Podfile env var)

leotm replied:
successfully reverted in https://github.com/MetaMask/metamask-mobile/pull/7114/commits/544cd1cf4c542a5335037a9a7f69ce94ca698353 as no longer required

tested working w Flipper and Xcode 14.2/15 and RN CLI from clean builds

leotm committed:
Bump Branch IPHONEOS_DEPLOYMENT_TARGETs from 8 to 12

leotm committed:
Bump MixPanel IPHONEOS_DEPLOYMENT_TARGETs from 8 to 12

leotm commented:
TODO: Bump

https://github.com/MetaMask/metamask-mobile/blob/30e2693941504c16328680959d8a2314444d84fd/bitrise.yml#L816-L819

from https://github.com/bitrise-io/stacks/blob/main/content/stack_reports/osx-xcode-14.2.x-ventura.md

to https://github.com/bitrise-io/stacks/blob/main/content/stack_reports/osx-xcode-15.0.x.md
or https://github.com/bitrise-io/stacks/blob/main/content/stack_reports/osx-xcode-15.0.x-edge.md

leotm committed:
Bump BitRise stack from osx-xcode-14.2.x-ventura to osx-xcode-15.0.x

jpcloureiro commented:
> nb: CI git submodules error and Xcode Sentry error still need a resolution


To solve the submodule errors we will have to either update BranchIO & Mixpanel submodules or fork their repos to apply the changes[1]

Changing the files inside these submodules directly will make the git submodule point to an invalid commit hash.
<img width="504" alt="image" src="https://github.com/MetaMask/metamask-mobile/assets/1649425/dc967ee8-3452-4671-a203-549df2a2fabb">

[1] [1c9fc2b](https://github.com/MetaMask/metamask-mobile/pull/7114/commits/1c9fc2b6dbfc918fad71a04252280bcda0137d7d) & [71a8b0e](https://github.com/MetaMask/metamask-mobile/pull/7114/commits/71a8b0e74a6c54f62383ab8eaf1dbf43e67fd743)

leotm committed:
Revert "Bump MixPanel IPHONEOS_DEPLOYMENT_TARGETs from 8 to 12"

This reverts commit 71a8b0e74a6c54f62383ab8eaf1dbf43e67fd743.

leotm committed:
Revert "Bump Branch IPHONEOS_DEPLOYMENT_TARGETs from 8 to 12"

This reverts commit 1c9fc2b6dbfc918fad71a04252280bcda0137d7d.

leotm committed:
Fix Podfile react_native_post_install whitespace indentation

leotm committed:
Disable Podfile custom react_native_post_install

Use default RN react_native_post_install

leotm committed:
Fix FlipperKit files not found

TODO: Revert this and figure why FB_SONARKIT_ENABLED isn't disabled via
- https://github.com/MetaMask/metamask-mobile/commit/1e432e05dc3ba74e9cd9d548686a39ad38b269ad
- https://github.com/MetaMask/metamask-mobile/commit/1e432e05dc3ba74e9cd9d548686a39ad38b269ad

leotm committed:
Fix FlipperKit use of undeclared identifiers

TODO: Revert this and figure why FB_SONARKIT_ENABLED isn't disabled via
- https://github.com/MetaMask/metamask-mobile/commit/1e432e05dc3ba74e9cd9d548686a39ad38b269ad
- https://github.com/MetaMask/metamask-mobile/commit/1e432e05dc3ba74e9cd9d548686a39ad38b269ad

leotm commented:
details on how to currently build locally
- https://github.com/getsentry/sentry-react-native/issues/3322#issuecomment-1750805368

leotm commented on bitrise.yml:
```diff
@@ -815,7 +815,7 @@ app:
 meta:
   bitrise.io:
-    stack: osx-xcode-14.2.x-ventura
+    stack: osx-xcode-15.0.x
```
worth considering: https://github.com/bitrise-io/stacks/blob/main/content/changelogs/osx-xcode-15.0.x-edge.md
> with more frequent tool updates

leotm replied:
*https://github.com/bitrise-io/stacks/blob/main/content/changelogs/osx-xcode-15.1.x-edge.md

leotm replied:
resolving as discussed, start on base/vanilla `15.0.x`

then bump in near future as required or for future-proofing

sethkfman commented on ios/Podfile:
```diff
@@ -20,43 +20,43 @@ if linkage != nil
-	  project.save()
-	end
-  end
+# def react_native_post_install(installer)
```
Can we add a comment to why all iOS Flipper config is being commented out? We could also just remove it and then have someone implement Flipper.

leotm replied:
then a seperate comment for disabling our custom `react_native_post_install` which fixes the Sentry pod issue

leotm replied:
comment added here: https://github.com/MetaMask/metamask-mobile/pull/7114/commits/4bb7c33891c53e53119bc99deee8d3ce9f36f893

leotm replied:
the root cause of
- `NO_FLIPPER=1` (script) / `ENV['NO_FLIPPER'] = "1"` (Podfile) env vars not fully disabling Flipper
  - encountered previously in https://github.com/MetaMask/metamask-mobile/pull/6955
- Flipper acting problematic with Xcode 15
- `FB_SONARKIT_ENABLED=0` (script) / `ENV['FB_SONARKIT_ENABLED'] = "0"` (Podfile) env vars not disabling Flipper
- Flipper enabled in release mode

turned out to be `FB_SONARKIT_ENABLED=1` still currently forced on via Apple Clang compiler flags

leotm committed:
Update mixpanel-iphone git submodule commit hash

- Bump MixPanel IPHONEOS_DEPLOYMENT_TARGETs from 8 to 12
- On branch bump-iphoneios-deployment-targets-from-8-to-12

jpcloureiro committed:
update .gitmodule file with mixpanel-iphone fork. remove duplicate submodule

leotm committed:
Update mixpanel-iphone git submodule commit hash

- Bump MixPanel IPHONEOS_DEPLOYMENT_TARGETs from 8 to 12
- On branch: bump-branch-ios-sdk-deployment-targets-from-8-to-12

leotm committed:
update .gitmodule file with branch-ios-sdk fork. remove duplicate submodule

leotm commented:
tested working also on old Xcode 14.2

tommasini commented:
Hey @leotm I see that the next version of react native ([0.71.14](https://github.com/facebook/react-native/releases/tag/v0.71.14)), have a xcode 15 patch

leotm committed:
Update podfile.lock

leotm commented on .gitmodules:
```diff
@@ -1,10 +1,6 @@
 	path = ios/branch-ios-sdk
-	url = https://github.com/BranchMetrics/ios-branch-deep-linking-attribution
-
-[submodule "ios/mixpanel-iphone"]
-	path = ios/mixpanel-iphone
-	url = https://github.com/mixpanel/mixpanel-iphone
```
this was a duplicate stale submodule left hanging around, removed as tech debt

leotm commented on ios/branch-ios-sdk:
fork and branch created in use here:

https://github.com/MetaMask/ios-branch-deep-linking-attribution/tree/bump-branch-ios-sdk-deployment-targets-from-8-to-12

leotm committed:
Remove duplicate Podfile flipper_config comment

leotm committed:
Remove stale Podfile flipper_post_install comment

- Last used: https://github.com/facebook/react-native/blame/v0.63.5/template/ios/Podfile#L21-L23
- Removed: https://github.com/facebook/react-native/blob/v0.64.0/template/ios/Podfile
- Moved from Podfile to scripts/react_native_pods.rb: https://github.com/facebook/react-native/commit/574447ae3d6bf15f29a8c4a64b2355373f2bb1bc

leotm committed:
Refactor Podfile react_native_post_install

leotm committed:
Reenable Podfile flipper_config

leotm committed:
Move Podfile flipper_config comment to correct location

Since it was originally in global scope

leotm committed:
Remove FB_SONARKIT_ENABLED flag from xcodeproj release build settings

Should have been added only as debug (not release) config in our RN 0.62.2 upg:
https://github.com/MetaMask/metamask-mobile/pull/1586

Ref: https://github.com/facebook/react-native/blob/v0.62.0/template/ios/HelloWorld.xcodeproj/project.pbxproj#L500

leotm committed:
Refactor xcodeproj FB_SONARKIT_ENABLED flag to macro

Should have been added as an Apple Clang preprocessor macro (not compiler flag) in our RN 0.62.2 upg:
https://github.com/MetaMask/metamask-mobile/pull/1586

Refactor in xcodeproj target
- from: Apple Clang - Custom Compiler Flags > Other C Flags
- to: Apple Clang - Preprocessing > Preprocessor Macros

Ref: https://github.com/facebook/react-native/blob/v0.62.0/template/ios/HelloWorld.xcodeproj/project.pbxproj#L500

Nb: rn-tester sticking to "-DFB_SONARKIT_ENABLED=1" for OTHER_CFLAGS and OTHER_SWIFT_FLAGS

leotm committed:
Remove FB_SONARKIT_ENABLED flag from xcodeproj debug build settings

Originally added in our RN 0.62.2 upg:
https://github.com/MetaMask/metamask-mobile/pull/1586

Should have been removed since RN 0.63.0:
https://github.com/facebook/react-native/blob/v0.63.0/template/ios/HelloWorld.xcodeproj/project.pbxproj

Ref: https://github.com/facebook/react-native/pull/28796

leotm commented on ios/MetaMask.xcodeproj/project.pbxproj:
```diff
@@ -1137,14 +1097,11 @@
 				ONLY_ACTIVE_ARCH = NO;
-				OTHER_CFLAGS = (
-					"$(inherited)",
-					"-DFB_SONARKIT_ENABLED=1",
```
details: https://github.com/MetaMask/metamask-mobile/pull/7114/commits/aaba386b5cba0aa2ff6621910f311838bb32d917

leotm commented on ios/MetaMask.xcodeproj/project.pbxproj:
```diff
@@ -1071,14 +1034,11 @@
 				ONLY_ACTIVE_ARCH = YES;
-				OTHER_CFLAGS = (
-					"$(inherited)",
-					"-DFB_SONARKIT_ENABLED=1",
```
details
- https://github.com/MetaMask/metamask-mobile/pull/7114/commits/83be066351574bf308eb48836fa2950798abfb3e
- https://github.com/MetaMask/metamask-mobile/pull/7114/commits/a2d850f6e9dc29a49672a0eed1c0793f3847e18e

leotm committed:
Reenable AppDelegate.m Flipper imports

- at top of file globally
- within initializeFlipper fn

leotm committed:
Remove NO_FLIPPER=1 from package.json setup script

leotm committed:
Add back Flipper to Podfile and xcodeproj

leotm commented on react-native.config.js:
```diff
@@ -2,6 +2,9 @@
 // eslint-disable-next-line import/no-commonjs
 module.exports = {
   dependencies: {
+    ...(process.env.NO_FLIPPER
+      ? { 'react-native-flipper': { platforms: { ios: null } } }
+      : {}),
```
see [Podfile comment](https://github.com/MetaMask/metamask-mobile/pull/7114/files#diff-281ded35b124f5160d0a57e47e521ee6f49f233933e091a0d33f3fd42a74abc8R49-R57) for details

leotm committed:
Revert "Fix Xcode 15 error: Unable to load ...frameworks-Debug-input-files.xcfilelist"

This reverts commit a09d517135af02474bbef3e461a7e50d32f99830.

leotm commented on .gitmodules:
```diff
@@ -1,10 +1,6 @@
+	url = https://github.com/metamask/mixpanel-iphone
 [submodule "ios/branch-ios-sdk"]
 	path = ios/branch-ios-sdk
-	url = https://github.com/BranchMetrics/ios-branch-deep-linking-attribution
-
-[submodule "ios/mixpanel-iphone"]
-	path = ios/mixpanel-iphone
-	url = https://github.com/mixpanel/mixpanel-iphone
+	url = https://github.com/metamask/ios-branch-deep-linking-attribution
```
nb: if the fork creator doesn't have sufficient permissions,
git will error 403 attempting to fetch it (permission denied)

<img width="847" alt="Screenshot 2023-10-11 at 4 42 11 pm" src="https://github.com/MetaMask/metamask-mobile/assets/1881059/2c82f269-ad5c-4e89-8b33-a711b82dc1b5">

spotted one way to verify this is - check for the `Settings` tab (cc @jpcloureiro)

<img width="1009" alt="Screenshot 2023-10-16 at 10 27 44 am" src="https://github.com/MetaMask/metamask-mobile/assets/1881059/ace0ff39-9dd4-46d6-ab03-ae47b540d885">

leotm committed:
Add Podfile react_native_post_install comment

leotm commented:
> Hey @leotm I see that the next version of react native ([0.71.14](https://github.com/facebook/react-native/releases/tag/v0.71.14)), have a xcode 15 patch, I thought it was good to post here

great shout @tommasini ^ RE
> Xcode 15 patch ([287482e57f](https://github.com/facebook/react-native/commit/287482e57ffd221227e6fffb6852113d330260a1) by [@fortmarek](https://github.com/fortmarek))

we encountered the issue https://github.com/facebook/react-native/issues/37748 earlier in https://github.com/MetaMask/metamask-mobile/issues/6158#issuecomment-1701340482 and https://github.com/MetaMask/metamask-mobile/issues/6158#issuecomment-1743375058

was initially applying the fix https://github.com/facebook/react-native/issues/37748#issuecomment-1580334650 locally in Xcode

but interestingly not needed to the past couple weeks running clean builds on RN 0.71.13 with current changes

but you're right let's complete the job and bump to RN 0.71.14 so it doesn't arise again

i'll do the same for Detox bumping from 20.11.0 to 20.11.4 to include (cc @cortisiko)
> [20.11.2] iOS: ensure compatibility with Xcode 15 beta (by asafkorem in https://github.com/wix/Detox/pull/4152).

leotm commented:
TODO: fix BitRise smoke tests failing on Xcode 15

https://app.bitrise.io/build/6645117d-458d-49e1-b99e-4f6af3a162b5

```console
# All tests failing due to unrecognised device
# ...
FAIL e2e/specs/../x.spec.js
  ● Test suite failed to run
    DetoxRuntimeError: Failed to find a device by type = "iPhone 12 Pro"
    HINT: Run 'applesimutils --list' to list your supported devices. It is advised only to specify a device type, e.g., "iPhone Xʀ" and avoid explicit search by OS version.
      at SimulatorAllocDriver._queryDevices (node_modules/detox/src/devices/allocation/drivers/ios/SimulatorAllocDriver.js:115:13)
      at SimulatorAllocDriver._groupDevicesByStatus (node_modules/detox/src/devices/allocation/drivers/ios/SimulatorAllocDriver.js:94:27)
      at SimulatorAllocDriver._findOrCreateDevice (node_modules/detox/src/devices/allocation/drivers/ios/SimulatorAllocDriver.js:81:29)
      at node_modules/detox/src/devices/allocation/drivers/ios/SimulatorAllocDriver.js:31:14
      at safeAsync (node_modules/detox/src/utils/safeAsync.js:4:13)
      at node_modules/detox/src/devices/DeviceRegistry.js:73:24
      at ExclusiveLockfile.exclusively (node_modules/detox/src/utils/ExclusiveLockfile.js:41:15)
      at SimulatorAllocDriver.allocate (node_modules/detox/src/devices/allocation/drivers/ios/SimulatorAllocDriver.js:30:18)

# ...
```

leotm commented:
> TODO: fix BitRise smoke tests failing on Xcode 15
> 
> https://app.bitrise.io/build/6645117d-458d-49e1-b99e-4f6af3a162b5
> 
> ```
> # All tests failing due to unrecognised device
> # ...
> FAIL e2e/specs/../x.spec.js
>   ● Test suite failed to run
>     DetoxRuntimeError: Failed to find a device by type = "iPhone 12 Pro"
>     HINT: Run 'applesimutils --list' to list your supported devices. It is advised only to specify a device type, e.g., "iPhone Xʀ" and avoid explicit search by OS version.
>       at SimulatorAllocDriver._queryDevices (node_modules/detox/src/devices/allocation/drivers/ios/SimulatorAllocDriver.js:115:13)
>       at SimulatorAllocDriver._groupDevicesByStatus (node_modules/detox/src/devices/allocation/drivers/ios/SimulatorAllocDriver.js:94:27)
>       at SimulatorAllocDriver._findOrCreateDevice (node_modules/detox/src/devices/allocation/drivers/ios/SimulatorAllocDriver.js:81:29)
>       at node_modules/detox/src/devices/allocation/drivers/ios/SimulatorAllocDriver.js:31:14
>       at safeAsync (node_modules/detox/src/utils/safeAsync.js:4:13)
>       at node_modules/detox/src/devices/DeviceRegistry.js:73:24
>       at ExclusiveLockfile.exclusively (node_modules/detox/src/utils/ExclusiveLockfile.js:41:15)
>       at SimulatorAllocDriver.allocate (node_modules/detox/src/devices/allocation/drivers/ios/SimulatorAllocDriver.js:30:18)
> 
> # ...
> ```

able to reproduce this locally

<img width="1903" alt="detox xcode 15" src="https://github.com/MetaMask/metamask-mobile/assets/1881059/cef421d3-680e-4208-bb88-e8e4d906e372">

leotm committed:
Upgrade React Native from 0.71.13 to 0.71.14

https://react-native-community.github.io/upgrade-helper/?from=0.71.13&to=0.71.14

Tested working on both
- Xcode 14.2
- Xcode 15

leotm committed:
Bump Detox from 20.11.0 to 20.11.4

Includes 20.11.2 Xcode 15 compat

leotm committed:
Merge branch 'main' into react-native-0.71.13-xcode-15-beta-8

cortisiko commented:
@leotm did you create the iPhone 12 Pro simulator? Seems like detox cannot find the simulator.

leotm committed:
Remove unneeded extra spaces in build scripts

leotm committed:
Bump Detox iOS sim from iPhone 12 Pro to iPhone 13 Pro for Xcode 15

leotm committed:
Merge branch 'main' into react-native-0.71.13-xcode-15-beta-8

leotm commented:
@SocketSecurity ignore @metamask/oss-attribution-generator@2.0.1

leotm commented on .detoxrc.js:
```diff
@@ -56,7 +56,7 @@ module.exports = {
       type: 'ios.simulator',
       device: {
-        type: 'iPhone 12 Pro',
+        type: 'iPhone 13 Pro',
```
see https://github.com/MetaMask/metamask-mobile/pull/7114#issuecomment-1768403708

leotm replied:
atm locally if Xcode 15 hasn't been opened and an older iPhone 12 Pro sim manually created

the default Xcode 15 iPhone 13 Pro will launch instead

leotm replied:
TODO: consider which of these need updating now to iPhone 13 Pro (cc @cortisiko)

<img width="537" alt="Screenshot 2023-10-18 at 5 42 30 pm" src="https://github.com/MetaMask/metamask-mobile/assets/1881059/e5c8cafb-4c6b-40d9-8ace-45a9516994c3">

leotm replied:
build scripts updated (following up slack convo)
- https://github.com/MetaMask/metamask-mobile/pull/7114/commits/5006b1ad0411cd4814f503616da676f2c12411f6
- https://github.com/MetaMask/metamask-mobile/pull/7114/commits/c066fc014d19f93dcac7ad5da5f52523001ed73a

again happy for us to update these to vanilla `iPhone 13`s instead

leotm replied:
also updated our Browserstack configs in: https://github.com/MetaMask/metamask-mobile/pull/7114/commits/8e5c7d5fabb7b67b8ff9f7de8a7f43704c75314d to finish the job (no more `iPhone 12 Pro` refs in codebase)

should we choose to run `test:wdio:ios` in BitRise (now on stack `osx-xcode-15.0.x`) again in the future

jpcloureiro replied:
For some reason a fresh Xcode 15 install on a fresh macOS Ventura 13.6 (i386 arch) only contains iPhone 15 variants, failing with the following error.
![Screenshot 2023-10-25 at 17 58 05](https://github.com/MetaMask/metamask-mobile/assets/1649425/08cd395e-b594-47f5-9e8f-6f6f9099de46)



Wouldn't consider this a blocker though.

jpcloureiro replied:
After adding the requested simulator, it compiles & runs the app.

![Screenshot 2023-10-25 at 18 22 38](https://github.com/MetaMask/metamask-mobile/assets/1649425/30332adc-4b00-438c-a1af-86b7775e6e1d)

Intel based macs should have no problems

leotm replied:
i'll add a note on this in our ReadMe after merge

thanks for testing again ^

leotm commented on ios/Podfile:
```diff
@@ -4,58 +4,61 @@ require_relative '../node_modules/@react-native-community/cli-platform-ios/nativ
+
+# 	arm_value = `/usr/sbin/sysctl -n hw.optional.arm64 2>&1`.to_i
+
+# 	projects.each do |project|
+# 	  project.build_configurations.each do |config|
+# 		  if arm_value == 1 then
+# 		    config.build_settings.delete("EXCLUDED_ARCHS[sdk=iphonesimulator*]")
+# 		  else
+# 		    config.build_settings["EXCLUDED_ARCHS[sdk=iphonesimulator*]"] = "arm64"
+# 		  end
+# 	  end
+
+# 	  project.save()
```
hopefully @Cal-L this pr works on your Intel Mac running Xcode 15

but if not we can look at reenabling this part of our custom `react_native_post_install`

(hopefully without resurfacing the Sentry issue)

jpcloureiro replied:
Tested on macOS Ventura 13.6 with an Intel chip  https://github.com/MetaMask/metamask-mobile/pull/7114/files#r1372094036

leotm committed:
Update build scripts from iPhone 12 Pro to iPhone 13 Pro

And match current Detox sim passing BitRise smoke tests

iPhone 13 Pro comes setup ootb on Xcode 15

iPhone 13 is the default launched by RN

leotm committed:
Update release build scripts from iPhone 12 Pro to iPhone 13 Pro

Follow-up: https://github.com/MetaMask/metamask-mobile/pull/7114/commits/5006b1ad0411cd4814f503616da676f2c12411f6

Despite us not yet running Detox in release mode due to issues

leotm committed:
Update ReadMe iPhone 12 Pro refs to iPhone 13 Pro

leotm committed:
Update ReadMe Xcode 14.2 to Xcode 15.0

leotm committed:
Update ReadMe Xcode 15.0 to Xcode 15.0.1

leotm commented on package.json:
```diff
@@ -446,7 +446,7 @@
     "lint-staged": "10.5.4",
     "metro": "0.72.3",
-    "metro-react-native-babel-preset": "0.73.9",
+    "metro-react-native-babel-preset": "0.73.10",
```
good Qu ^ we [used to](https://github.com/facebook/react-native/blame/f3c98c5fa273dbdb17c32cd1193a3f078ef942b4/template/package.json) caret then [pinned](https://github.com/facebook/react-native/blame/ba5454c42134cc09772f024772b19a9990b1fb7f/template/package.json) after lengthy [discussion](https://github.com/facebook/react-native/pull/34803#pullrequestreview-1122165575) to ensure

leotm replied:
> Plot twist:
> 
> [facebook/react-native#41219](https://github.com/facebook/react-native/pull/41219)

🤪 ♾️

leotm commented on package.json:
```diff
@@ -446,7 +446,7 @@
     "koa": "^2.14.2",
     "lint-staged": "10.5.4",
     "metro": "0.72.3",
```
good spot i can't see a reason to be tracking it in package.json at all :suspect:

leotm replied:
can confirm this have never offically been tracked:
https://react-native-community.github.io/upgrade-helper/?from=0.23.1&to=0.71.14#RnDiffApp-package.json

leotm replied:
either way good spot again can confirm we should be on the ver you suggested 👌

verified by `npx react-native@0.71.14 init RN07114 --version 0.71.14`

```console
➜  RN07114 npm list metro
RN07114@0.0.1 /Users/leo/Documents/GitHub/RN07114
└─┬ react-native@0.71.14
  └─┬ @react-native-community/cli@10.2.4
    └─┬ @react-native-community/cli-plugin-metro@10.2.3
      ├─┬ metro-config@0.73.10
      │ └── metro@0.73.10 deduped
      └─┬ metro@0.73.10
        └─┬ metro-transform-worker@0.73.10
          └── metro@0.73.10 deduped
```

however ideally pinned for same reasons as: https://github.com/MetaMask/metamask-mobile/pull/7114#discussion_r1371875954

which we can also see in
- https://github.com/facebook/metro/blob/main/packages/metro-config/package.json#L19
- https://github.com/facebook/metro/blob/main/packages/metro-transform-worker/package.json#L20

so added the commit here: https://github.com/MetaMask/metamask-mobile/pull/7114/commits/8fbec65ce8f1cd3eea2cc60398e7e51be00c292a

leotm replied:
great explanation was about to add the pr too ^
and now the circular deps seem v obvious 🤪

otherwise not addressing this complicates future RN upgrades
and risks breaking our metro config or RN CLI from incompat vers

re-thinking let's go couple last steps further
- <s>move `metro-config` from _dep_ to _devDep_</s>
- (more thought) untrack it in package.json

(so now fully/officially aligned on [RN core](https://github.com/facebook/react-native/blob/v0.71.14/template/package.json#L28) w simplest setup possible)

then cleanup

https://github.com/MetaMask/metamask-mobile/blob/3022144b736727b8f15536d90a067087f85860ed/.depcheckrc.yml#L37

i'll add comment on your pr

leotm committed:
Bump metro devDep from 0.72.3 to 0.73.10

See: https://github.com/MetaMask/metamask-mobile/pull/7114#pullrequestreview-1696481657

leotm committed:
Remove stale Device util static method isIphone12

Tech debt: Device.isIphone12 is unused

leotm committed:
Bump Browserstack iPhone 12 Pro devices to iPhone 13 Pro

leotm committed:
Merge branch 'main' into react-native-0.71.13-xcode-15-beta-8

leotm committed:
Fix lint error remove unused var

legobeat committed:
Merge branch 'main' into react-native-0.71.13-xcode-15-beta-8

jpcloureiro reviewed (approved):
lgtm

cortisiko commented:
@legobeat One small comment, please update the new default device in the Readme: https://github.com/MetaMask/metamask-mobile#ios and https://github.com/MetaMask/metamask-mobile#ios-2

leotm committed:
Merge branch 'main' into react-native-0.71.13-xcode-15-beta-8

leotm commented:
https://app.bitrise.io/app/be69d4368ee7e86d/pipelines/e773b76c-7813-4ab2-bbc2-240093da1797

[ios_e2e_test](https://app.bitrise.io/build/5122d22c-adf6-4bcf-9563-78ba249ff60b) ✅

[android_e2e_test](https://app.bitrise.io/build/7056c20a-ad55-434c-9cb8-6887d0157632) ❌

unsure how the pr affects Android :suspect:
likely flakey test, going to re-run

leotm commented:
> https://app.bitrise.io/app/be69d4368ee7e86d/pipelines/e773b76c-7813-4ab2-bbc2-240093da1797
> 
> [ios_e2e_test](https://app.bitrise.io/build/5122d22c-adf6-4bcf-9563-78ba249ff60b) ✅
> 
> [android_e2e_test](https://app.bitrise.io/build/7056c20a-ad55-434c-9cb8-6887d0157632) ❌
> 
> unsure how the pr affects Android :suspect: likely flakey test, going to re-run

yeah likely flakey BitRise internet connection

first Android failing on `2 network requests` :suspect:

```console
BUILD SUCCESSFUL in 16m 18s
998 actionable tasks: 993 executed, 5 up-to-date
17:20:22.986 detox[34293] B jest --config e2e/jest.e2e.config.js --testNamePattern Smoke
17:20:32.014 detox[34351] i onboarding-wizard-opt-in.spec.js is assigned to emulator-5554 (emulator)
17:20:32.025 detox[34351] i (node:34351) V8: /bitrise/src/node_modules/ganache/dist/node/0.js:2 Linking failure in asm.js: Unexpected stdlib member
(Use `node --trace-warnings ...` to show where the warning was created)
17:20:51.639 detox[34351] i typed-sign.spec.js is assigned to emulator-5554 (emulator)
17:20:51.648 detox[34351] i (node:34351) V8: /bitrise/src/node_modules/ganache/dist/node/0.js:2 Linking failure in asm.js: Unexpected stdlib member
17:21:16.078 detox[34351] i add-custom-rpc.spec.js is assigned to emulator-5554 (emulator)
17:21:16.087 detox[34351] i (node:34351) V8: /bitrise/src/node_modules/ganache/dist/node/0.js:2 Linking failure in asm.js: Unexpected stdlib member
17:21:20.313 detox[34351] i permission-system-delete-wallet.spec.js is assigned to emulator-5554 (emulator)
17:21:20.317 detox[34351] i (node:34351) V8: /bitrise/src/node_modules/ganache/dist/node/0.js:2 Linking failure in asm.js: Unexpected stdlib member
17:21:25.251 detox[34351] i swap-tokens.spec.js is assigned to emulator-5554 (emulator)
17:21:25.258 detox[34351] i (node:34351) V8: /bitrise/src/node_modules/ganache/dist/node/0.js:2 Linking failure in asm.js: Unexpected stdlib member
17:21:30.871 detox[34351] i browser-tests.spec.js is assigned to emulator-5554 (emulator)
17:21:30.903 detox[34351] i (node:34351) V8: /bitrise/src/node_modules/ganache/dist/node/0.js:2 Linking failure in asm.js: Unexpected stdlib member
17:21:37.945 detox[34345] i start-exploring.spec.js is assigned to emulator-15646 (emulator)
17:21:37.974 detox[34345] i (node:34345) V8: /bitrise/src/node_modules/ganache/dist/node/0.js:2 Linking failure in asm.js: Unexpected stdlib member
(Use `node --trace-warnings ...` to show where the warning was created)
17:21:43.221 detox[34351] i Starting fixture server...
17:21:43.225 detox[34351] i The fixture server is started
17:21:43.225 detox[34351] i Loading JSON state...
17:21:43.228 detox[34351] i JSON state loaded
17:21:53.886 detox[34345] i The app is busy with the following tasks:
• 2 network requests with URLs:
  - URL #1: https://api.etherscan.io/api?module=account&offset=40&sort=desc&action=txlist&tag=latest&page=1
  - URL #2: https://api.etherscan.io/api?module=account&offset=40&sort=desc&action=tokentx&tag=latest&page=1
17:22:03.715 detox[34351] i The app is busy with the following tasks:
• "LooperIdlingResource-380-mqt_js" (JS Thread) is executing (JavaScript code).
• 2 network requests with URLs:
  - URL #1: https://api.etherscan.io/api?module=account&address=0x76cf1CdD1fcC252442b50D6e97207228aA4aefC3&offset=40&sort=desc&action=tokentx&tag=latest&page=1
  - URL #2: https://api.etherscan.io/api?module=account&address=0x76cf1CdD1fcC252442b50D6e97207228aA4aefC3&offset=40&sort=desc&action=txlist&tag=latest&page=1
• Activity on the React-Native bridge.
17:22:03.893 detox[34345] i The app is busy with the following tasks:
• 2 network requests with URLs:
  - URL #1: https://api.etherscan.io/api?module=account&offset=40&sort=desc&action=txlist&tag=latest&page=1
  - URL #2: https://api.etherscan.io/api?module=account&offset=40&sort=desc&action=tokentx&tag=latest&page=1
17:22:13.725 detox[34351] i The app is busy with the following tasks:
• 2 network requests with URLs:
  - URL #1: https://api.etherscan.io/api?module=account&address=0x76cf1CdD1fcC252442b50D6e97207228aA4aefC3&offset=40&sort=desc&action=tokentx&tag=latest&page=1
  - URL #2: https://api.etherscan.io/api?module=account&address=0x76cf1CdD1fcC252442b50D6e97207228aA4aefC3&offset=40&sort=desc&action=txlist&tag=latest&page=1
17:22:13.901 detox[34345] i The app is busy with the following tasks:
• 2 network requests with URLs:
  - URL #1: https://api.etherscan.io/api?module=account&offset=40&sort=desc&action=txlist&tag=latest&page=1
  - URL #2: https://api.etherscan.io/api?module=account&offset=40&sort=desc&action=tokentx&tag=latest&page=1
FAIL e2e/specs/settings/addressbook-tests.spec.js
  ● Test suite failed to run
    Exceeded timeout of 120000ms while setting up Detox environment
17:22:48.615 detox[34345] i The app is busy with the following tasks:
• "LooperIdlingResource-360-mqt_js" (JS Thread) is executing (JavaScript code).
• Activity on the React-Native bridge.
17:23:46.217 detox[34345] i The app seems to be idle
PASS e2e/specs/wallet/start-exploring.spec.js (207.037 s)
```

leotm commented:
> > https://app.bitrise.io/app/be69d4368ee7e86d/pipelines/e773b76c-7813-4ab2-bbc2-240093da1797
> > [ios_e2e_test](https://app.bitrise.io/build/5122d22c-adf6-4bcf-9563-78ba249ff60b) ✅
> > [android_e2e_test](https://app.bitrise.io/build/7056c20a-ad55-434c-9cb8-6887d0157632) ❌
> > unsure how the pr affects Android :suspect: likely flakey test, going to re-run
> 
> yeah likely flakey BitRise internet connection
> 
> first Android failing on `2 network requests` :suspect:
> 

yep it was flakey BitRise internet on those tests, Android now passing

https://app.bitrise.io/build/9e7f91de-ec83-4075-afdc-3933dd427853

[android_e2e_test](https://app.bitrise.io/build/9e7f91de-ec83-4075-afdc-3933dd427853) ✅
